### ISSUE TITLE
Display number of pages from AXIS in processing team queues

### ISSUE THREAD

**Issue by arielleandrews** [bcgov/foi-flow#1684]
* As a processing team analyst
* I want the total number of pages in a request to display in my queue 
* so that I can effectively manage my requests and prioritize my work

**Assumptions & Scope**	
* This user story requires sync with AXIS
* Number of pages will pull from the "Total # of pages in a request" field in AXIS 
* Processing queue will be the only one with this field; will not show up in other queues
* Field will display from "Open" onwards
* Will be able to sort/filter based on contents of the field

What is IN scope?
* Populating the Pages fields for a request with the value following sync with AXIS

What is NOT in scope?
* Pages column in processing team queue is already complete
* Updates to processing team queue covered in user story #1672

**Acceptance Criteria**	

**Scenario 1: Sync with AXIS - field is blank**
* GIVEN a processing team analyst is logged in
* WHEN the analyst syncs a request with AXIS
* AND the "Total # of pages in a request" field in AXIS is blank
* THEN the "Pages" field in the processing team queue will display as a blank space in the column 
* AND the sync with AXIS button will be inactive

**Scenario 2: Sync with AXIS - field is populated**
* GIVEN a processing team analyst is logged in
* WHEN the "Total # of pages in a request" field in AXIS is populated with a number
* THEN the sync with AXIS button will be active
* AND the banner will appear along the top indicating the request is out of sync

**Scenario 3: Sync with AXIS - update populated field**
* GIVEN a processing team analyst is logged in
* WHEN the analyst syncs a request with AXIS
* THEN the sync with AXIS button will be active
* AND the banner will appear along the top indicating the request is out of sync

**Scenario 4: Sync with AXIS - Sync Review Changes**
* GIVEN a processing team analyst has activated the sync button when there is a change to total # of pages
* WHEN the modal opens
* THEN the changes to be reviewed will also include the total number of pages

**Scenario 5: Sync with AXIS - Complete Sync**
* GIVEN a processing team analyst is reviewing the changes to be synced
* WHEN the analyst confirms the change
* THEN the value for total # of pages in a request will be displayed in their queue for the request they synced.

**Dependencies? What is the impact of this dependency? (If so, link dependency in the ticket, make it visible in a team´s backlog)**

**Validation Rules? (If yes, list here)**	

**Design**
@xxx - please link the Design here

**Definition of Ready**

1. [ ] Is there a well articulated User Story?
2. [ ] Is there Acceptance Criteria that covers all scenarios (happy/sad paths)?
3. [ ] If there is a user interface, is there a design? 
4. [ ] Does the user story need user research/validation?
5. [ ] Does this User Story needs stakeholder approval?
6. [ ] Design / Solution accepted by Product Owner
7. [ ] Is this user story small enough to be completed in a Sprint? Should it be split?
8. [ ] Are the dependencies known/ understood? (technical, business, regulatory/policy)
9. [ ] Has the story been estimated?

**Definition of Done**

1.    [ ] Passes developer unit tests
2.    [ ] Passes peer code review
3.    [ ] If there's a user interface, passes UX assurance
4.    [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
5.	[ ] Confirm Test cases built and succeeding 
6.	[ ] No regression test failures
7.	[ ] Test coverage acceptable by Product Owner
8.	[ ] Ticket ready to be merged to master or story branch
9.	[ ] Developer to list Config changes/ Update documents and designs
10.	[ ] Can be demoed in Sprint Review 
11.	[ ] Tagged as part of a Release
12.   [ ] Feature flagged if required
13.  [ ] Change Management activities done?

**Comment by lmullane** [comment]
Looks good to me, @arielleandrews

**Comment by m-prodan** [comment]
@divyav-aot @aparna-aot 

I believe you both worked on this one? It does not appear that I can test scenario 3, I can't add pages to an existing file right now. 

Regarding scenario 2 - I may have found a bug. When there are multiple folders in document management, it only appears to count pages from one of the folders. I am not sure if its an issue with that particular file, or if we are pulling the wrong value? The file I used was FNR-2021-15044 - When I run the report for the field "Total # of pages in a request" - I can get the correct number of 2 pages, but once I sync it in the new system it only displays 1 page. That particular request had 2 folders though, and each had a single page. My other single-folder test cases worked as expected.

**Comment by m-prodan** [comment]
@divyav-aot @aparna-aot 

I think this one may need to go back for further review. One of the issues is that there was no prompt or ability to sync the file to show the number of pages. Only if I added an extension, or made a change to the request details could I sync the file and then have the number of pages displayed.

When total # of pages has a value other than nil/0 - it should trigger a banner warning and allow for a sync. Should I update the AC or can you fix it with this comment? Let me know - moving back to review for your comments.

cc: @lmullane @KaraBeach @mpilchar

**Comment by divyav-aot** [comment]
@m-prodan we can update the banner display and mdal for total no of pages. Please updated the AC.

**Comment by m-prodan** [comment]
Blocked until FOI-MOD74 (AXIS bugs) are addressed by IMB - unable to add or remove pages to new or existing requests

**Comment by mpilchar** [comment]
@m-prodan Is this unblocked now that we are on 10.3?

**Comment by m-prodan** [comment]
Yup - going to try and tackle this one before I call it a day :)

**Comment by m-prodan** [comment]
QA passed - you all have no idea how excited IAO is going to be seeing this in action, I can't wait to demo this at the branch meeting on tuesday! Great work @divyav-aot
